src/fold.jl

```diff
@@ -0,0 +1,111 @@
+
+function unfold_kernel!(col::AbstractArray{T}, x, col_size, input_size, output_size, kernel_size, flipkernel, stride, pad_lo, dilation, max_idx) where {T}
+    index = threadIdx().x + (blockIdx().x - 1) * blockDim().x
+
+    @inbounds if index <= max_idx
+        i, kw, kh, kd, c, b = CartesianIndices(col_size)[index].I # col indices
+        w, h, d = CartesianIndices(output_size)[i].I # x indices
+
+        # project 
+        w, h, d = @. ((w, h, d) - 1)*stride - pad_lo + 1 + ((kw, kh, kd) - 1)*dilation
+
+        if !flipkernel
+            kw, kh, kd = kernel_size .- (kw, kh, kd) .+ 1
+        end
+
+        # check out of bounds
+        if any((w, h, d) .<= 0 .|| (w, h, d) .> input_size)
```

Comment:
Buildkite is complaining this syntax isn't 1.6 compatible. Maybe try
```suggestion
        if any((w, h, d) .<= 0 .| (w, h, d) .> input_size)
```

Reply:
Ah sorry, looks like `|` might bind more tightly. The good news is that I found that the stdlib has `Base.checkbounds` and `Base.checkindex`, so the current conditional logic could be simplified.

Reply:
nice find, tyty.